test/unit/Model.js

```diff
@@ -1476,6 +1476,161 @@ describe("Model", () => {
+							}
+						});
+						const user = await callType.func(User).bind(User)(1);
+						expect(user.toJSON()).to.eql({
+							"id": 1,
+							"name": "Charlie",
+							"parent": 2
+						});
+						expect(getItemTimesCalled).to.eql(1);
+					});
+
+					it("Should not populate document automatically if schema property is object", async () => {
+						let getItemTimesCalled = 0;
+
+						User = dynamoose.model("User", {"id": Number, "name": String, "parent": {"type": dynamoose.model("Parent", {"id": Number, "data": String})}});
+						dynamoose.aws.ddb.set({
+							"getItem": (params) => {
+								getItemTimesCalled++;
+								return {"promise": () => params.Key.id.N === "1" ? {"Item": {"id": {"N": "1"}, "name": {"S": "Charlie"}, "parent": {"N": "2"}}} : {"Item": {"id": {"N": "2"}, "name": {"S": "Bob"}}}};
+							}
+						});
+						const user = await callType.func(User).bind(User)(1);
+						expect(user.toJSON()).to.eql({
+							"id": 1,
+							"name": "Charlie",
+							"parent": 2
+						});
+						expect(getItemTimesCalled).to.eql(1);
+					});
+
+					it("Should not populate document automatically when schema property is dynamoose.THIS", async () => {
+						let getItemTimesCalled = 0;
+
+						User = dynamoose.model("User", {"id": Number, "name": String, "parent": dynamoose.THIS});
+						dynamoose.aws.ddb.set({
+							"getItem": (params) => {
+								getItemTimesCalled++;
+								return {"promise": () => params.Key.id.N === "1" ? {"Item": {"id": {"N": "1"}, "name": {"S": "Charlie"}, "parent": {"N": "2"}}} : {"Item": {"id": {"N": "2"}, "name": {"S": "Bob"}}}};
+							}
+						});
+						const user = await callType.func(User).bind(User)(1);
+						expect(user.toJSON()).to.eql({
+							"id": 1,
+							"name": "Charlie",
+							"parent": 2
+						});
+						expect(getItemTimesCalled).to.eql(1);
+					});
+
+					it("Should not populate document automatically when schema property is dynamoose.THIS if schema property is object", async () => {
+						let getItemTimesCalled = 0;
+
+						User = dynamoose.model("User", {"id": Number, "name": String, "parent": {"type": dynamoose.THIS}});
+						dynamoose.aws.ddb.set({
+							"getItem": (params) => {
+								getItemTimesCalled++;
+								return {"promise": () => params.Key.id.N === "1" ? {"Item": {"id": {"N": "1"}, "name": {"S": "Charlie"}, "parent": {"N": "2"}}} : {"Item": {"id": {"N": "2"}, "name": {"S": "Bob"}}}};
+							}
+						});
+						const user = await callType.func(User).bind(User)(1);
+						expect(user.toJSON()).to.eql({
+							"id": 1,
+							"name": "Charlie",
+							"parent": 2
+						});
+						expect(getItemTimesCalled).to.eql(1);
+					});
+
+					// TODO: reeneable the tests below for populate with sets
+					/*
+					it("Should not populate document automatically when using set", async () => {
+						let getItemTimesCalled = 0;
```

Comment:
Based on this, I am assuming the sets/arrays use case is not going to be part of this ticket?

Reply:
@BraidenCutforth There was an issue here. I forget exactly what it was. Let me circle back and see what I can do.